pip/index.py

```diff
@@ -842,18 +856,37 @@ def clean_link(self, url):
 
 class Link(object):
 
-    def __init__(self, url, comes_from=None):
+    def __init__(self, url, comes_from=None, requires_python=None):
+        """
+        Object representing a parsed link from https://pypi.python.org/simple/*
+
+        url:
+            url of the resource pointed to (href of the link)
+        comes_from:
+            <Not sure>
+        requires_python:
+            String containing the `Requires-Python` metadata field, specified
+            in PEP 345.
+        """
 
         # url can be a UNC windows share
         if url.startswith('\\\\'):
             url = path_to_url(url)
 
         self.url = url
         self.comes_from = comes_from
+        if not requires_python:
+            self.requires_python = None
+        else:
+            self.requires_python = requires_python
```

Comment:
nitpicking but `self.requires_python = requires_python if requires_python else None` should fit in the 79 characters